File: missingRanges.py
## Given a sorted integer array where the ranges of elements are [start, end] inclusive,
## return its missing ranges 

def missingRanges(nums, start, end):
  output = []
  nums.insert(0, start - 1)
  nums.append(end + 1)
  
  for i in range(len(nums) - 1):
    count = len(range(nums[i], nums[i + 1]))
    if(count == 2):
      output.append(str(nums[i] + 1))
    elif(count > 2):
      temp = str(nums[i] + 1) + '->' + str(nums[i + 1] - 1)
      output.append(temp)
      
  return output 

File: Array/MissingRanges.java
import java.util.ArrayList;
import java.util.List;

/*
 * Given a sorted integer array where the range of the elements are [start, end] inclusive, 
 * Return its missing ranges. 
 */
 
class Solution {
	
	public List<String> findMissingRanges(int[] array, int start, int end ){
		List<String> output = new ArrayList<String>(); 
		int curr, prev  = start - 1 ; 
		for(int i = 0; i <= array.length; i ++ ){
			curr = (i == array.length )? end + 1 : array[i];
			if(curr - prev == 2)output.add("" + (prev + 1)); 
			if(curr - prev > 2) output.add("" + (prev + 1) + "->" + (curr - 1)); 
			prev = curr; 
		}
		return output; 
	}
	
	public static void main(String[] args) {
		
		int[][] testCase = {{2, 4, 5, 50, 75}, {}}; 
		int start = 0, end = 99; 
		for(int[] array: testCase){
			System.out.println(new Solution().findMissingRanges(array, start, end ));
		}
	}
}
